lib/src/block_parser.dart

```diff
@@ -713,9 +721,36 @@ abstract class ListSyntax extends BlockSyntax {
     var anyEmptyLinesBetweenBlocks = false;
 
     for (var item in items) {
-      var itemParser = BlockParser(item.lines, parser.document);
+      bool hasCheck = false, checked;
+      if (item.lines.isNotEmpty) {
+        final m = _reCheckbox.firstMatch(item.lines[0]);
+        if (hasCheck = m != null) {
+          checked = m[1] == 'x';
+          item.lines[0] = m[2];
+        }
+      }
+
+      var itemParser = BlockParser(item.lines, parser.document, item.offset);
       var children = itemParser.parseLines();
-      itemNodes.add(Element('li', children));
+      if (hasCheck) {
+        final check = Element('input', null);
+        if (checked) check.attributes['checked'] = 'checked';
+        if (parser.document.checkable)
+          check.attributes['data-line'] = '${item.offset}';
```

Comment:
I don't see this `data-line` feature on GitHub. I don't think we should add features or complexity beyond what a popular client (like GitHub or CommonMark) has.

Reply:
When using a checklist, the app needs to modify the markdown when user clicks on the checkbox. By generating the line number, it is easier for app to write such kind of code. Or, we need another complicated utility to locate `[ ]` from the markdown.